### PR TITLE
Expose MultMatrix

### PR DESCRIPTION
**Pull request by rocktavious**
This adds the MultMatrix with is in raylib core so you can easily setup rendering matrix instead of having to make several calls to `Translatef` and `Rotatef` and `Scalef`.

### CODE REVIEW

**Comment by rocktavious** [comment]
I have tested this in my project and it works properly.

**Comment by gen2brain** [comment]
Can you please also add that function to rlgl_purego.go

File: raylib/rlgl_purego.go [hunk 1]
```diff
@@ -16,6 +16,7 @@ var rlLoadIdentity func()
 var rlTranslatef func(x float32, y float32, z float32)
 var rlRotatef func(angle float32, x float32, y float32, z float32)
 var rlScalef func(x float32, y float32, z float32)
+var rlMultMatrix func(m []float32)
```
Comment (rocktavious) [inline raylib/rlgl_purego.go:19]: @gen2brain should this be the function signature of `C.rlMultMatrixf` or of the golang function?  If the former then it should likely be Matrix not a []float32.

I'm not familiar with this purego stuff.

Comment (gen2brain) [reply raylib/rlgl_purego.go:19]: @rocktavious Yes, the C signature, is similar to `dlopen` or `libffi`. The Go functions should both be the same of course.

Comment (rocktavious) [reply raylib/rlgl_purego.go:19]: @gen2brain - so just for 100% clarity - what I have right now is good - i just need to add 1 function in `rlgl_purego` that is `func MultMatrix(m Matrix) {` ?

Comment (gen2brain) [reply raylib/rlgl_purego.go:19]: You need a pointer, like `var rlMultMatrix func(m *float32)`, and then pass `unsafe.Pointer(&m[0])`.

Comment (rocktavious) [reply raylib/rlgl_purego.go]: @gen2brain - ok i think i've done this correctly now.  I'm not sure how to actually test this to see if this works.  I've never used the purego implementation.

Comment (JupiterRider) [reply raylib/rlgl_purego.go]: @rocktavious 
```console
[jupiterrider@pc42 raylib]$ GOOS=windows go test 
# github.com/gen2brain/raylib-go/raylib
./rlgl_purego.go:305:7: undefined: MatrixToFloat
./rlgl_purego.go:306:2: undefined: rlMultMatrixf
```

Comment (rocktavious) [reply raylib/rlgl_purego.go]: @JupiterRider - the 2nd on makes sense that a typo on my part i'll fix but the first one

```
./rlgl_purego.go:305:7: undefined: MatrixToFloat
```

The function is just in the raylib-go library in https://github.com/gen2brain/raylib-go/blob/5314a4a2099ae5c4d32ad9147b5966b7e6486b5c/raylib/rcore.go#L858

So i'm not sure why its not finding that.  There is something i'm missing in my mental model of how all this pure go stuff works.

Comment (JupiterRider) [reply raylib/rlgl_purego.go]: @rocktavious 
I created a new pull request, so you can remove your purego implementation.

**Comment by JupiterRider** [comment]
@gen2brain 
Is our MatrixToFloat implementation correct?
There is also MatrixToFloatV.

```golang
// MatrixToFloatV - Get float array of matrix data
func MatrixToFloatV(mat Matrix) [16]float32 {
	var result [16]float32

	result[0] = mat.M0
	result[1] = mat.M1
	result[2] = mat.M2
	result[3] = mat.M3
	result[4] = mat.M4
	result[5] = mat.M5
	result[6] = mat.M6
	result[7] = mat.M7
	result[8] = mat.M8
	result[9] = mat.M9
	result[10] = mat.M10
	result[11] = mat.M11
	result[12] = mat.M12
	result[13] = mat.M13
	result[14] = mat.M14
	result[15] = mat.M15

	return result
}
```

```golang
// MatrixToFloat - Converts Matrix to float32 slice
func MatrixToFloat(mat Matrix) []float32 {
	data := make([]float32, 16)

	data[0] = mat.M0
	data[1] = mat.M4
	data[2] = mat.M8
	data[3] = mat.M12
	data[4] = mat.M1
	data[5] = mat.M5
	data[6] = mat.M9
	data[7] = mat.M13
	data[8] = mat.M2
	data[9] = mat.M6
	data[10] = mat.M10
	data[11] = mat.M14
	data[12] = mat.M3
	data[13] = mat.M7
	data[14] = mat.M11
	data[15] = mat.M15

	return data
}
```

C-raylib:
```c
#ifndef MatrixToFloat
    #define MatrixToFloat(mat) (MatrixToFloatV(mat).v)
#endif

RMAPI float16 MatrixToFloatV(Matrix mat)
{
    float16 result = { 0 };

    result.v[0] = mat.m0;
    result.v[1] = mat.m1;
    result.v[2] = mat.m2;
    result.v[3] = mat.m3;
    result.v[4] = mat.m4;
    result.v[5] = mat.m5;
    result.v[6] = mat.m6;
    result.v[7] = mat.m7;
    result.v[8] = mat.m8;
    result.v[9] = mat.m9;
    result.v[10] = mat.m10;
    result.v[11] = mat.m11;
    result.v[12] = mat.m12;
    result.v[13] = mat.m13;
    result.v[14] = mat.m14;
    result.v[15] = mat.m15;

    return result;
}

```

**Comment by gen2brain** [comment]
@JupiterRider Sorry, I don't remember what happened there.

**Comment by rocktavious** [comment]
Thanks @JupiterRider and @gen2brain  - should be good now.

**Pull request merged**